presentation/src/main/java/com/smile/presentation/MainActivityFragment.kt

```diff
@@ -43,11 +41,15 @@ class MainActivityFragment : BaseFragment() {
     }
 
     private fun initViews() {
-        characterAdapter =
-            CharacterAdapter(characterItems, ::onImageLoaded, ::transitionToDetailView)
+        characterAdapter = CharacterAdapter(
+            viewModel.characterItems,
```

Comment:
Hmmm... Note that any change to this item in the adapter will update that of the viewmodel. But it has its advantages, as well as disadvantages, but it looks pretty great and interesting right now :)